### PR TITLE
collin/cleanup

### PR DESCRIPTION
**Pull request by cwebbles**
Add cleanup of workflows after completion

### CODE REVIEW

File: src/quest/historian.py [hunk 1]
```diff
@@ -832,6 +832,7 @@ async def _func(*a, **kw):
                     assert record['task_id'] == task_id
 
             logging.debug(f'Completing task {task_id}')
+            self._history.clear()
```
Comment (cwebbles) [inline src/quest/historian.py:835]: We wanted the Workflow Manager to have control of the cleaning process, but the test we talked about didn't include one. Also, should the cleaning be as close to the history as possible? Is there a case where there isn't a workflow manager?

File: src/quest/manager.py [hunk 2]
```diff
@@ -66,6 +68,7 @@ def _get_workflow(self, workflow_id: str):
         return self._workflows[workflow_id]
 
     def _remove_workflow(self, workflow_id: str):
+        self._workflows.get(workflow_id)._history.clear()
```
Comment (cwebbles) [inline src/quest/manager.py:71]: Hacky. Haven't tested this yet. Didn't seem like _remove_workflow is always called after a workflow is finished? Guessing_game didn't seem to run this.

File: src/quest/persistence.py [hunk 3]
```diff
@@ -52,6 +52,13 @@ def remove(self, item: EventRecord):
         self._storage.delete_blob(key)
         self._storage.write_blob(self._namespace, self._keys)
 
+    def clear(self):
+        for key in self._keys:
+            self._storage.delete_blob(key)
+        self._items.clear()
+        self._keys.clear()
+        self._storage.delete_blob(self._namespace)
+
```
Comment (cwebbles) [inline src/quest/persistence.py:61]: Anything more special that this? On the passing test this wiped out the records. All that was left was the base directory.

File: src/quest/historian.py (hunk 1 above)
Comment (cwebbles) [reply src/quest/historian.py:835]: After adding this a few tests are breaking. Again, not sure if this is the correct implementation.

File: src/quest/historian.py [hunk 4]
```diff
@@ -877,16 +877,19 @@ async def _run(self, *args, **kwargs):
...
             external_task.add_done_callback(lambda t: self._open_tasks.remove(t) if t in self._open_tasks else None)
 
-            return await self.start_task(
+            task = await self.start_task(
                 self._run_with_exception_handling,
                 *args, **kwargs,
                 name=f'{self.workflow_id}.main',
                 task_factory=_task_factory.create_task
             )
+            self._history.clear()
```
Comment (byubean) [inline src/quest/historian.py:886]: I think the `self._history.clear()` should go in `_run_with_exception_handling`.

Also, please move the `self._workflow_completed = True` and `await self._handle_resource...` from `_run_with_args` to `run_with_exception_handling`.

**Pull request merged**